source/sdk/threading/native/ThreadUnix.ooc

```diff
@@ -3,69 +3,56 @@ import ../[Thread], math, os/Time
 include unistd
 
 version(unix || apple) {
-
-	/**
-	 * pthreads implementation of threads.
-	 */
 	ThreadUnix: class extends Thread {
-
 		pthread: PThread
 
-		init: func ~unix (=_code) {}
-
+		init: func ~unix (=_code)
 		start: func -> Bool {
 			result := pthread_create(pthread&, null, _code as Closure thunk, _code as Closure context)
 			(result == 0)
 		}
-
 		wait: func -> Bool {
 			result := pthread_join(pthread, null)
 			(result == 0)
 		}
-
 		wait: func ~timed (seconds: Double) -> Bool {
-			version (apple || android) { return __fake_timedjoin(seconds) }
+			status := false
+			version (apple || android)
+				status = __fake_timedjoin(seconds)
 			else {
 				ts: TimeSpec
 				__setupTimeout(ts&, seconds)
 
 				result := pthread_timedjoin_np(pthread, null, ts&)
-				return (result == 0)
+				status = (result == 0)
 			}
-			false
+			status
 		}
-
 		cancel: func -> Bool {
+			status := false
 			version (!android)
-				return this alive?() && (pthread_cancel(this pthread) == 0)
-			false
+				status = this alive?() && (pthread_cancel(this pthread) == 0)
+			status
```

Comment:
Here you have no excuse for not returning `result`. Just rename the variable you added :)